mozjs/js/rust/src/magicdom/domquad.rs

```diff
@@ -20,6 +25,22 @@ magic_dom! {
     }
 }
 
+impl NumSlots for DOMQuad {
+    const NUM_SLOTS: u32 = <dompoint::DOMPoint as NumSlots>::NUM_SLOTS_PTR * 4;
```

Comment:
The NUM_SLOTS is more for inheritance. If a class is inherit from DOMQuad, it will have a offset of 4 slots before its own slots. But if a class just contain a DOMQuad, it will just place 1 slot for DOMQuad. NUM_SLOTS_PTR is for the object stored as ObjectValue.